src/apis/fetchCategories.ts

```diff
@@ -0,0 +1,43 @@
+import { Category } from "../types";
+
+export const fetchCategories = async (): Promise<Category[]> => {
+    return await fetch("http://localhost:3001/categories")
```

Comment:
NIT( فلسفة وممكن يكون اشي منطقي ) 
since this part is redundant `http://localhost:3001/`
i think its better to store it in a constant, so you manage it from single place, if we decide to change the port to 3000 for example, we change that in single place.

so 
const  apiUrl = "http://localhost:3001"

and the below will be fetch (`${baseUrl}/categires`)